src/main/java/org/example/NPC.java

```diff
@@ -1,13 +1,89 @@
 package org.example;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.Scanner;
+
 public class NPC {
 
+    private final Scanner sc = new Scanner(System.in);
+    private final List<String> hints = new ArrayList<>();
+    private final List<Puzzles> puzzles = new ArrayList<>();
+    private String name;
+    private String description;
+    private Random random;
+
+
+    public NPC() {
+        addHints();
+        addPuzzles();
+    }
+
+    public NPC(String name, String description) {
+        this.name = name;
+        this.description = description;
+        addHints();
```

Comment:
change from addHints and addPuzzles to add only a single hint and puzzle depending on which room the npc is in.
Get the hint from the hint switch in the Text class.